File: packages/logger-ios/Sources/Internal/Compressor.swift
import Foundation

protocol Compressor {
    var algorithm: String { get }
    func compress(_ input: Data) throws -> Data
}

struct NoopCompressor: Compressor {
    let algorithm = "none"

    func compress(_ input: Data) throws -> Data {
        input
    }
}

struct ZlibCompressor: Compressor {
    let algorithm = "zlib"

    func compress(_ input: Data) throws -> Data {
        if #available(macOS 10.15, iOS 13.0, *) {
            return try (input as NSData).compressed(using: .zlib) as Data
        }
        return input
    }
}

File: packages/logger-android/src/main/kotlin/com/codex/logger/internal/Compressor.kt
package com.codex.logger.internal

internal interface Compressor {
    val algorithm: String
    fun compress(input: ByteArray): ByteArray
}

internal object NoopCompressor : Compressor {
    override val algorithm: String = "none"

    override fun compress(input: ByteArray): ByteArray = input
}

internal object ZlibCompressor : Compressor {
    override val algorithm: String = "zlib"

    override fun compress(input: ByteArray): ByteArray {
        val output = java.io.ByteArrayOutputStream()
        java.util.zip.DeflaterOutputStream(output).use { stream ->
            stream.write(input)
        }
        return output.toByteArray()
    }
}
